sycl/include/CL/sycl/detail/clusm.hpp

```diff
@@ -62,6 +62,7 @@ class CLUSM {
     const void *BaseAddress = nullptr;
     size_t Size = 0;
     size_t Alignment = 0;
+    cl_device_id Device = nullptr;
```

Comment:
Who knows that this is a pointer?...
```
cl_device_id Device {};
```

Reply:
OpenCL. :)

Type | Description | API Type
-- | -- | --
_cl_platform_id * | The ID for a platform. | cl_platform_id

Reply:
That makes sense. :-)
But I searched again https://www.khronos.org/registry/OpenCL/specs/2.2/pdf/OpenCL_API.pdf and https://www.khronos.org/registry/OpenCL/specs/opencl-1.0.pdf about `cl_device_id` and there is no concrete definition.
I always thought the OpenCL data types were just opaque implementation-dependent data types ...
In the specification about some API, it is said that some values can be `NULL`.
Since OpenCL API is implemented in C and I guess `NULL` comes from `#include <stddef.h>`, this probably allows:
- all the integer types, if we do not care about warning;
- pointer types :-)

So on a small system with 1 device could be a `cl_device_id`... :-)

Interesting, it does not compile with `float` or `double`... :-)

I guess I missed something obvious somewhere...